llvm/lib/SYCLLowerIR/ESIMD/ESIMDUtils.cpp

```diff
@@ -123,5 +123,62 @@ Type *getVectorTyOrNull(StructType *STy) {
   return Res;
 }
 
+bool isInvokeSimdBuiltinCall(const CallInst *CI) {
+  Function *F = CI->getCalledFunction();
+
+  if (F && F->getName().startswith(INVOKE_SIMD_PREF)) {
+    return true;
+  }
+  return false;
+}
+
+// Tracks the use of a function pointer being stored in a memory.
+// Returns false if the function pointer is used as an argument for invoke_simd
+// function call, true otherwise.
+bool filterFunctionPointer(Value *address) {
```

Comment:
1) Since this function remains quite specific and is not reused anywhere, it should be moved into the place of use - LowerESIMDKernelAttrs.cpp.
2) Please improve comment. It is not clear what is memory, what is function pointer, what is the incoming argument.
3) I added a note below that this should have different signature.